worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/ContractAdapterStub.java

```diff
@@ -19,65 +18,117 @@
  */
 @Slf4j
 @Component
-public class BeneficiaryAdapterStub implements BeneficiaryAdapter {
+public class ContractAdapterStub implements ContractAdapter {
 
     private static final String BENE_ID_FILE = "/test-stub-data/synthetic-bene-ids.csv";
-
-    @Value("${bb-stub.page.size}")
-    private int pageSize;
-
+    private static final int MAX_ROWS = 30_000;
 
     @Override
-    public GetPatientsByContractResponse getPatientsByContract(String contractNumber) {
+    public GetPatientsByContractResponse getPatients(String contractNumber) {
 
         final int contractSno = extractContractSno(contractNumber);
-        final int startOffset = (contractSno - 1) * 100;
 
-        final var patientsPerContract = getFromSampleFile(startOffset);
+        final var patientsPerContract = fetchPatientRecords(contractSno);
 
         return toResponse(contractNumber, patientsPerContract);
     }
 
 
     private Integer extractContractSno(String contractNumber) {
-        final String contractNumberSuffix = contractNumber.substring(contractNumber.length() - 3);
+        //valid range from 0000 - 9999
+        final String contractNumberSuffix = contractNumber.substring(contractNumber.length() - 4);
 
-        final Integer sno;
+        Integer sno = null;
         try {
             sno = Integer.valueOf(contractNumberSuffix);
         } catch (NumberFormatException e) {
             log.warn("Invalid contractNumber : {} ", contractNumber);
             return 0;
         }
 
+        // simple check for contractNumber with 5 digits instead of 4
+        try {
+            final String tmpContractNo = contractNumber.substring(contractNumber.length() - 5);
+            int contractNo = Integer.valueOf(tmpContractNo);
+            if (sno != contractNo) {
+                sno = -1;
+            }
+        } catch (NumberFormatException e) {
+            //ignore - this check is to see if the contractNumber has 5 digits instead of 4 making it > 9999
+        }
+
         return sno;
     }
 
-    private List<String> getFromSampleFile(final int startOffset) {
-        if (startOffset < 0) {
+    private List<String> fetchPatientRecords(final int contractSno) {
+        if (contractSno < 0) {
             return new ArrayList<String>();
         }
 
+        int numberOfRows = determineNumberOfRows(contractSno);
+        int rowsToRetrieve = determineRowsToRetrieve(numberOfRows);
+
+        var patientIdRows = getFromSampleFile(rowsToRetrieve);
+        final List<String> allRows = new ArrayList<>();
+        allRows.addAll(patientIdRows);
+
+        if (numberOfRows > MAX_ROWS) {
+            allRows.addAll(createMoreRows(numberOfRows, rowsToRetrieve, patientIdRows));
+        }
+
+        return allRows;
+    }
+
+    private int determineNumberOfRows(int contractSno) {
+        if (contractSno == 0) {
+//            return 100;   // temporary change till BFD calls mocked
+            return 10;
```

Comment:
Temporary change for now, till the BFD calls are mocked - so that only 10 patient records are returned for `S0000`

Reply:
reverted, so that it now returns 100 patient records for contractNo = 0000